src/oslogin_utils.cc

```diff
@@ -1231,4 +1246,128 @@ bool ContinueSession(bool alt, const string& email, const string& user_token, co
 
   return ret;
 }
+
+static bool ApplyPolicy(const char *user_name, string email, const char *policy, struct AuthOptions opts) {
+  std::stringstream url;
+  url << kMetadataServerUrl << "authorize?email=" << UrlEncode(email) << "&policy=" << policy;
+
+  // Don't try to add fingerprint parameter to policy call if we don't have it.
+  if (opts.fp_len > 0) {
+    url << "&fingerprint=" << opts.fingerprint;
+  }
+
+  string response;
+  long http_code = 0;
+  // Invalid user, just leave from here - the principal will not be allowed/authorized.
+  if (!HttpGet(url.str(), &response, &http_code)) {
+    SysLogErr("Failed to validate organization user %s has login permission.", user_name);
+    return false;
+  }
+
+  if (http_code != 200) {
+    SysLogErr("Failed to validate organization user %s has login permission, "
+              "got HTTP response code: %lu", user_name, http_code);
+    return false;
+  }
+
+  if (!ParseJsonToSuccess(response)) {
+    SysLogErr("Organization user %s does not have login permission.", user_name);
+    return false;
+  }
+
+  return true;
+}
+
+static bool CreateGoogleUserFile(string users_filename) {
+  std::ofstream users_file;
+  users_file.open(users_filename.c_str());
```

Comment:
I'm not sure this is the best way to check for the existence of a file. Why not use `stat`?

Reply:
Well, it was moved from the previous implementation, as the original author didn't document the decision making for it I can only guess why it's like that, probably they wanted to avoid using posix calls(?), since we are not very much taking much care of having it cross platform elsewhere I'd say it's safe to use `stat` - it will probably be even faster for obvious reasons. I can change it.

Reply:
Thanks, Leo. It seems like stat is used elsewhere in this codebase anywhere, so this shouldn't limit our compatibility any further than it already is. :)

Reply:
We were actually using `stat()` to check if file exists, I updated the function to make it more clear what it's actually doing, also introduced a `FileExists()` function using `access` instead.